### PR TITLE
Add more data to getPedAnimation

### PR DESCRIPTION
**Pull request by Dezash**
Adds additional return data to getPedAnimation function (time, loop, updatePosition, interruptable, freezeLastFrame, blendTime, restoreTaskOnAnimEnd) in the same order as setPedAnimation requires.

This is useful because when having lots of resources that set animations, it becomes complicated to manage them. For example, a server might have an animation panel which allows users to set any animation and another resource that sets an uninterruptible animation which isn't suppose to be stopped/changed by the user. Instead of setting element data or otherwise saving the state of that animation, you could get it from getPedAnimation.

Test resource:
[getPedAnimation.zip](https://github.com/multitheftauto/mtasa-blue/files/3119945/getPedAnimation.zip)

### CODE REVIEW

File: Client/mods/deathmatch/logic/CClientPed.cpp [hunk 1]
```diff
@@ -5789,6 +5789,37 @@ const char* CClientPed::GetAnimationName()
...
+{
+    return m_bFreezeLastFrameAnimation;
+}
+
+const int CClientPed::GetAnimationBlendTime()
+{
+    return m_iBlendAnimation;
+}
+
+
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientPed.cpp:5808]:
```suggestion
```

**Comment by qaisjp** [comment]
Seems reasonable. An alternative would be to have `getPedAnimationData` that returns a table. Which is better?

**Comment by Dezash** [comment]
I think it is better to keep it in getPedAnimation because adding another function that does the same thing with some extra features would make getPedAnimation redundant. However, I think it would make sense to implement a boolean function IsPedDoingAnimation to have a more efficient way to check if a ped is doing an animation.

**Comment by qaisjp** [comment]
> I think it is better to keep it in getPedAnimation because adding another function that does the same thing with some extra features would make getPedAnimation redundant.

Very true.

> However, I think it would make sense to implement a boolean function IsPedDoingAnimation to have a more efficient way to check if a ped is doing an animation.

Why do you think this is a more efficient way?

**Comment by Dezash** [comment]
> Why do you think this is a more efficient way?

Having less return data would be somewhat faster. Although the difference is negligible so I'm not sure if there's an actual need for a separate function.

```
function moreReturnData()
return "string1", "string2", false, true, true, false, 450, 4354
end

function lessReturnData()
return true
end

local a = os.clock()
for i=1, 10000000 do
lessReturnData()
end
print(os.clock() - a)
a = os.clock()
for i=1, 10000000 do
moreReturnData()
end
print(os.clock() - a)
```

Result:
0.29
0.38

**Comment by qaisjp** [comment]
How about the third arg being a table containing the extra data?

I dislike how setPedAnimation currently takes many arguments (hard to read) and would like to avoid the returns having the same problem.

**Comment by Dezash** [comment]
Done

**Comment by ghost** [comment]
It's better to define a new struct for animation data instead of using multiple member variables in CClientPed.

**Comment by ghost** [comment]
I didn't know that you addressed the changes. There are still a few improvements that could be made. I'll do it myself.

**Pull request merged**